packages/drivers/odsp-driver/src/odspDocumentServiceFactory.ts

```diff
@@ -35,3 +40,70 @@ export class OdspDocumentServiceFactory
         );
     }
 }
+
+export class LocalOdspDocumentServiceFactory extends OdspDocumentServiceFactoryCore {
+    constructor(
+        private readonly fluidFile: Uint8Array | string,
+    ) {
+        super(
+            async (_options) => { return null; }, // TODO
+            undefined,
+            async () => getSocketIo(), // TODO
+            undefined,
+            undefined, // TODO
+        );
+    }
+
+    public async createContainer(
+        _createNewSummary: ISummaryTree | undefined,
+        _createNewResolvedUrl: IResolvedUrl,
+        logger?: ITelemetryBaseLogger,
+        _clientIsSummarizer?: boolean,
+    ): Promise<IDocumentService> {
+        const odspLogger = createOdspLogger(logger);
+
+        return this.createDocumentServiceCore(
+            this.getFakeOdspResolvedUrl(),
+            odspLogger,
+        );
+    }
+
+    protected async createDocumentServiceCore(
+        resolvedUrl: IResolvedUrl,
+        odspLogger: TelemetryLogger,
+        _cacheAndTrackerArg?: ICacheAndTracker,
+        _clientIsSummarizer?: boolean,
+    ): Promise<IDocumentService> {
+        return new LocalOdspDocumentService(this.getFakeOdspResolvedUrl(), odspLogger, this.fluidFile);
+    }
```

Comment:
I prefer defensive programming :)
I'd assert here that
- _clientIsSummarizer !== true
- _cacheAndTrackerArg === undefined
- resolvedUrl: We either set expectation that it's some dummy value (and assert it here), or we assume that it actually contains info about specific file (then call to this.getFakeOdspResolvedUrl() should be removed and we pass actual URL).